### PR TITLE
Melodic style changes

### PR DESCRIPTION
**Pull request by wjwwood**
This includes the cherry-picked (and then deconflicted) commits from https://github.com/ros/pluginlib/pull/64 as well as ABI breaking style changes.

### CODE REVIEW

File: include/pluginlib/pluginlib_exceptions.h [hunk 1]
```diff
@@ -39,63 +40,69 @@ namespace pluginlib
  * @class PluginlibException
  * @brief A base class for all pluginlib exceptions that inherits from std::runtime_exception
  */
-class PluginlibException: public std::runtime_error
+class PluginlibException : public std::runtime_error
 {
-  public:
-    PluginlibException(const std::string & error_desc) : std::runtime_error(error_desc) {}
+public:
+public:
```
Comment (mikaelarguedas) [inline include/pluginlib/pluginlib_exceptions.h]: copy-n-paste ?

**Review by mikaelarguedas — COMMENTED** [review]
this is not an actual review, I just noticed something fishy on one of the commits. I'll review #64 first and then will review this once the bulk of the change has been merged into kinetic-devel

File: include/pluginlib/pluginlib_exceptions.h (hunk 1 above)
Comment (wjwwood) [reply include/pluginlib/pluginlib_exceptions.h]: Yeah, likely a problem from when I was de-conflicting the cherry-picking.

**Comment by wjwwood** [comment]
ready for re-review

**Comment by mikaelarguedas** [comment]
Thanks @wjwwood for iterating on this. I'm going to forward port #64 to the melodic-devel branch and then rebase and merge this

**Comment by wjwwood** [comment]
@mikaelarguedas that's what I meant to do, sorry, let me refactor this pr again.

**Comment by wjwwood** [comment]
I guess we'd need https://github.com/ros/pluginlib/pull/68 too?

**Comment by mikaelarguedas** [comment]
#68 has been forward ported already in #69 (except the changes to plugin_tool given that we removed it in melodic) so we should be good with #64 + the ABI breaking changes in this PR

**Comment by wjwwood** [comment]
Ok, I had to revert the forward port of #68 and then apply the forward port of style, then reapply it. If you squash and merge it won't show up. The only "new" commit is 511741a.

**Comment by mikaelarguedas** [comment]
Thanks!

**Pull request merged**